### PR TITLE
Experiment/daisy UI - refactored

### PR DESCRIPTION
**Pull request by silviot**
- Adds a static mock of the customer order UI built with daisy UI to showcase it. Check it out at `/orders/c/1`
- Updates superforms to v2 (partially, enough so that I can add a new CustomerOrderForm, and everything still works.... more to do here though)
- Updates prettier to v3 - which allows us to use newer svelte syntax

Supersedes #565

## Release Notes

- **New Features**
  - Introduced the `PageCenterDialog` component for enhanced modal dialog functionality.
  - Added `CustomerOrderMetaForm` for managing customer order metadata.
  
- **Improvements**
  - Enhanced form handling and validation across various components using `sveltekit-superforms` and `zod`.
  - Updated dependency versions for improved performance and compatibility.
  
- **Bug Fixes**
  - Minor formatting adjustments to improve code readability and consistency across multiple components.
  
- **Chores**
  - Updated linting scripts to simplify commands and improve the development workflow.

### CODE REVIEW

**Comment by silviot** [comment]
@coderabbitai full review

The above is in the hope that it will provide "Accept changes" buttons to accept its changes.
As far as teaching it, I believe answering its comments should be enough for it to store a fact in its "learnings" space.

**Comment by silviot** [comment]
@delasous I realized I @-addressed the AI but I meant to communicate with you.

It took me more than ideal to find out these facts, so I'll state them here. You might already be aware of them.

* The review should happen in the `files` tab, i.e. https://github.com/librocco/librocco/pull/566/files for this PR
* For each AI (or human!) suggestion one should choose between:
  * Ignore it and go on with your day
  * Click `Committable suggestion` and `Add suggestion to batch`
  * Reply to the bot to let it know things it doesn't know yet. It should save them somewhere for future reference
* When done click the `Commit suggestions` button at the top. Cross your fingers while waiting for the CI run.

File: apps/web-client/src/routes/inventory/inbound/[id]/+page.svelte [hunk 1]
```diff
@@ -485,12 +487,12 @@
...
 					<!-- {$connectivity} -->
 					<BookForm
-						data={bookFormData}
+						data={defaults(bookFormData, zod(bookSchema))}
 						publisherList={$publisherList}
 						options={{
 							SPA: true,
 							dataType: "json",
-							validators: bookSchema,
+							validators: zod(bookSchema),
```
[the remark this answers is not in the document]
Comment (delasous) [reply apps/web-client/src/routes/inventory/inbound/[id]/+page.svelte:495]: This is incorrect. v2 of superforms requires that a validation adapter, e.g `zod` wrap any schema before it is passed to `validators` or `defaults`. A data object does not need to be passed to `defaults` if there is no initial data

File: apps/web-client/src/routes/stock/+page.svelte [hunk 2]
```diff
@@ -251,12 +253,12 @@
...
 				<div class="px-6">
 					<BookForm
-						data={bookFormData}
+						data={defaults(bookFormData, zod(bookSchema))}
 						publisherList={$publisherList}
 						options={{
 							SPA: true,
 							dataType: "json",
-							validators: bookSchema,
+							validators: zod(bookSchema),
```
[the remark this answers is not in the document]
Comment (delasous) [reply apps/web-client/src/routes/stock/+page.svelte:261]: This is incorrect

**Pull request closed without merging**